### DISCUSSION TITLE
Range queries don't transmit other headers on Requests

### DISCUSSION THREAD

**Discussion by srijan55** [cortexproject/cortex#4410] [Q&A]
Hi,  I am using Cortex in  FrontEnd only mode, talking to a prometheus service serving the queries. But I found that the roundtrip tripperware , which takes care of aligning and splitting queries, strips of any headers which I specify on my queries and hence they don't reach my prometheus service. 
Are there known reasons for why this approach was chosen. I see that apart from OrgId all the headers are removed (including authorization headers)

**Comment by bboreham** [comment]
Pretty sure we just didn't think about it.
The frontend was written for Cortex, then quickly modified to be usable against non-Cortex services.

**Reply by srijan55** [reply]
Let me see if I can make a contribution for this.
